format/pyrdp/pyrdp.go

```diff
@@ -0,0 +1,144 @@
+// Processes PyRDP replay files
+// https://github.com/GoSecure/pyrdp
+//
+// Copyright (c) 2022-2023 GoSecure Inc.
+// Copyright (c) 2024 Flare Systems
+//
+// Maintainer: Olivier Bilodeau <olivier.bilodeau@flare.io>
+// Author: Lisandro Ubiedo
+package pyrdp
+
+import (
+	"time"
+
+	"github.com/wader/fq/format"
+	pyrdp "github.com/wader/fq/format/pyrdp/pdu"
+	"github.com/wader/fq/pkg/decode"
+	"github.com/wader/fq/pkg/interp"
+	"github.com/wader/fq/pkg/scalar"
+)
+
+const (
+	READ_EXTRA = true
+
+	// PDU Types.
+	PDU_FAST_PATH_INPUT            = 1  // Ex: scan codes, mouse, etc.
+	PDU_FAST_PATH_OUTPUT           = 2  // Ex: image
+	PDU_CLIENT_INFO                = 3  // Creds on connection
+	PDU_SLOW_PATH_PDU              = 4  // For slow-path PDUs
+	PDU_CONNECTION_CLOSE           = 5  // To advertise the end of the connection
+	PDU_CLIPBOARD_DATA             = 6  // To collect clipboard data
+	PDU_CLIENT_DATA                = 7  // Contains the clientName
+	PDU_MOUSE_MOVE                 = 8  // Mouse move event from the player
+	PDU_MOUSE_BUTTON               = 9  // Mouse button event from the player
+	PDU_MOUSE_WHEEL                = 10 // Mouse wheel event from the player
+	PDU_KEYBOARD                   = 11 // Keyboard event from the player
+	PDU_TEXT                       = 12 // Text event from the player
+	PDU_FORWARDING_STATE           = 13 // Event from the player to change the state of I/O forwarding
+	PDU_BITMAP                     = 14 // Bitmap event from the player
+	PDU_DEVICE_MAPPING             = 15 // Device mapping event notification
+	PDU_DIRECTORY_LISTING_REQUEST  = 16 // Directory listing request from the player
+	PDU_DIRECTORY_LISTING_RESPONSE = 17 // Directory listing response to the player
+	PDU_FILE_DOWNLOAD_REQUEST      = 18 // File download request from the player
+	PDU_FILE_DOWNLOAD_RESPONSE     = 19 // File download response to the player
+	PDU_FILE_DOWNLOAD_COMPLETE     = 20 // File download completion notification to the player
+)
+
+// TODO: Fill all descriptions.
+var pduTypesMap = scalar.UintMap{
```

Comment:
Can use `scalar.UintMapSymStr` to skip description

Reply:
Fixed with 0c69df62